spec/components/intervention_spec.rb

```diff
@@ -9,6 +9,12 @@ def component_name
     render_component({})
     assert_select ".gem-c-intervention__title", text: "Check the next steps for your limited company"
     assert_select ".gem-c-intervention__paragraph", text: "You might be interested in this because you’ve been browsing guidance relevant to starting a limited company."
-    assert_select ".gem-c-intervention__paragraph", text: "Hide this suggestion if it’s not relevant to you"
+    assert_select ".gem-c-intervention__paragraph a", text: "Hide this suggestion"
```

Comment:
This of course checks that the correct tracking attributes are present, which is fine, but I wonder if there should also be a javascript test to verify that the Analytics events are actually firing as expected. 

I think there are some examples you can have a look at [here](https://github.com/alphagov/govuk_publishing_components/tree/master/spec/javascripts/components).

Reply:
Actually to be fair you don't have much custom tracking here (just the event that's expected to fire on page load and the link click events I suppose) so it _might_ be overkill to add JS test coverage?  not actually sure.

Reply:
Every component that has a JS file also needs to have a JS test file. It should be fairly straightforward - spyon the GA tracking function, initialise the component, check that the GA tracking function was called.

Reply:
Added